rules/aip0158/response_plural_first_field.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0158
+
+import (
+	"fmt"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var responsePluralFirstField = &lint.MessageRule{
+	Name:   lint.NewRuleName(158, "response-plural-first-field"),
+	OnlyIf: isPaginatedResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// Rule check: for all messages that end in Response and contain a next_page_token field.
+		// Throw a linter warning if, the first field in the message is not named according to plural(message_name.to_snake().split('_')[1:-1]).
+		nextPageToken := m.FindFieldByName("next_page_token")
+		if nextPageToken == nil {
+			return nil
+		}
+
+		firstField := m.GetFields()[0]
+		pluralize := pluralize.NewClient()
```

Comment:
This leads us to a new question about pluralization and where English meets Engineering. If the resource is weird (e.g., person or cactus), which is correct?

Option A:
  - `GetCactus`, `ListCacti` (with `repeated Cactus cacti = 1`)
  - `GetPerson`, `ListPeople` (with `repeated Person people = 1`)

or Option B:
  - `GetCactus`, `ListCactuses` (with `repeated Cactus cactuses = 1`)
  - `GetPerson`, `ListPersons` (with `repeated Person persons = 1`)

Do we want to express an opinion on this? The code here says "Cactus (singular) -> Cacti (plural)" even if folks wanted the other way ("Cactuses").

Reply:
I prefer option A. However, the current library doesn't support it. I will find a new lib.

Reply:
I believe it does based on the example here: https://github.com/gertd/go-pluralize#word-make-plural

```
pluralize -word Cactus -cmd Plural

Plural(Cactus)     => Cacti
```

If we do go this route, it's important to have tests for these Latin- and Greek-origin cases (Cactus -> Cacti, Octopus -> Octopuses? Octopi? Octopedes?). As well as the weird English ones (e.g., Person -> People, Goose -> Geese, etc).

Reply:
Done